### DISCUSSION TITLE
How to define a link renderer and preserve renderer for relative references?

### DISCUSSION THREAD

**Discussion by mrwunderbar666** [Witiko/markdown#273] [Q&A]
Hi,

I've been trying to figure this out, but I am stuck. I would like to change the default behaviour of link renderering.

Suppose I have the following settings of the markdown package

```latex
\usepackage[footnotes=true,
            underscores=false,
            codeSpans=true,
            definitionLists,
            smartEllipses,
            fencedCode,
            contentBlocks,
            pipeTables,
            tableCaptions,
            relativeReferences=true,
            citations]{markdown}
```

And I have a link in my text as follows

```markdown
This is my [link](https://tex.stackexchange.com/questions/599526/rendering-autolinks-with-markdown-package/599779).

And here is Figure <#fig:bla>. 
```

Then the default behaviour is to turn the URL into a footnote and the figure reference to a clickable number.

I would like to change that so it mimics the default behaviour of Github or pandoc. The URL is hidden and the text becomes clickable.

I tried to define the settings as following:

```latex
\markdownSetup{
  renderers = {
    link = {\href{#3}{#1}}
  }
}
```

This works as intended, however, it breaks cross references to figures and tables. These disappear after the change and are rendered verbatim. Something like "And here is Figure #fig:bla"

How can I preserve the default behaviour for relative references?

**Comment by Witiko** [comment]
@mrwunderbar666 To change how direct and indirect links are rendered without chaning how relative references are handled, you can redefine [the `\markdownLaTeXRendererDirectOrIndirectLink` command][2] from [the default definitions of link renderer prototypes for LaTeX][1]:

``` tex
\renewcommand\markdownLaTeXRendererDirectOrIndirectLink[4]{%
  \href{#3}{#1}%
}
```

However, the default definitions may change at any time. Specifically, the `\markdownLaTeXRendererDirectOrIndirectLink` command may be removed in the future. Therefore, you are advised to copy [all default definitions for links][1] to your document if your document should compile a few years from now.

 [1]: https://github.com/Witiko/markdown/blob/36290d5c75687462734a3d7fac29b4bdd90659a5/markdown.dtx#L28741-L28816
 [2]: https://github.com/Witiko/markdown/blob/36290d5c75687462734a3d7fac29b4bdd90659a5/markdown.dtx#L28815-L28816
